include/veriblock/stateless_validation.hpp

```diff
@@ -7,18 +7,35 @@
 #include "veriblock/entities/btcblock.hpp"
 #include "veriblock/entities/vbkblock.hpp"
 #include "veriblock/entities/vtb.hpp"
+#include "veriblock/signutil.hpp"
 #include "veriblock/time.hpp"
 
 namespace VeriBlock {
 
-bool checkProofOfWork(const BtcBlock& block);
-
-bool checkProofOfWork(const VbkBlock& block);
-
 template <typename BlockType>
-bool checkMaximumDrift(const BlockType& block) {
+bool checkMaximumDrift(const BlockType& block, ValidationState& state) {
   uint32_t currentTime = currentTimestamp4();
-  return (uint32_t)block.timestamp < currentTime + ALLOWED_TIME_DRIFT;
+  if ((uint32_t)block.timestamp >= currentTime + ALLOWED_TIME_DRIFT) {
+    return state.Invalid("checkMaximumDrift()",
```

Comment:
Why not return ValidationState instead of bool?

Reply:
@blackyblack during chained validation, you will need to properly manage the validation state instead, and when you return something other than bool, you need to properly pass all error messages... which is not convenient